src/elementary_functions.jl

```diff
@@ -32,13 +32,19 @@ function exp_itensornetwork(
   k=default_k_value(),
   a=default_a_value(),
   c=default_c_value(),
-  dim::Int=default_dimension(),
+  dim::Int=default_dim(),
 )
   ψ = const_itensornetwork(s)
   Lx = length(dimension_vertices(ψ, dim))
   for v in dimension_vertices(ψ, dim)
-    sind = only(inds(s, v))
-    ψ[v] = ITensor(exp(a / Lx) * exp.(k * index_values_to_scalars(s, sind)), inds(ψ[v]))
+    sinds = inds(s, v)
+    sinds_dim = filter(i -> dimension(s, i) == dim, sinds)
+    sinds_not_dim = filter(i -> dimension(s, i) != dim, sinds)
+    linds = setdiff(inds(ψ[v]), sinds)
+    ψ[v] = prod([
```

Comment:
I think it becomes a bit complicated that way because this is constructing an `ITensor` of arbitrary number of physical indices. So it's easiest to write it as an outer product over those indices rather than trying to construct the array first and splat `sinds_dim`